src/main/java/com/github/kongchen/swagger/docgen/reader/AbstractReader.java

```diff
@@ -501,6 +546,21 @@ void processOperationDecorator(Operation operation, Method method) {
             extension.decorateOperation(operation, method, chain);
         }
     }
+    
+    protected String getOperationId(Method method, String httpMethod) {
+        if (isUseEnhancedOperationId()) {
+            return method.getDeclaringClass().getSimpleName() + "_" + method.getName() + "_" + httpMethod;
```

Comment:
Since that format will probably not suite everyone, what about changing `useEnhancedOperationId` from boolean to String `OperationIdFormat`, and define a few formatting options like `${methodName}`, `{httpMethod}`, `${className}`, `${packageName}`, and let the user define the format? So like `${className}_${methodName}_${httpMethod}`? So this would be much more flexible, and avoid issues/PRs that someone prefers `-` than `_`. ;-) WDYT?

Reply:
Ok probably ${} something is bad because it conflicts with maven variables ... so just something else, but you get the idea.

Reply:
Good idea. I will update my PR.